File: simple-stack-example-mvp/src/main/java/com/zhuinden/simplestackdemoexamplemvp/application/CustomApplication.java
package com.zhuinden.simplestackdemoexamplemvp.application;

import android.app.Application;
import android.content.Context;

import com.zhuinden.simplestackdemoexamplemvp.application.injection.DaggerSingletonComponent;
import com.zhuinden.simplestackdemoexamplemvp.application.injection.SingletonComponent;

/**
 * Created by Owner on 2017. 01. 26..
 */

public class CustomApplication extends Application {
    SingletonComponent singletonComponent;

    @Override
    public void onCreate() {
        super.onCreate();
        singletonComponent = DaggerSingletonComponent.create();
    }

    public static CustomApplication get(Context context) {
        return (CustomApplication)context.getApplicationContext();
    }

    public SingletonComponent getComponent() {
        return singletonComponent;
    }
}

File: examples/simple-stack-example-mvp-fragments/src/main/java/com/zhuinden/simplestackdemoexamplefragments/application/CustomApplication.kt
package com.zhuinden.simplestackdemoexamplefragments.application

import android.app.Application
import android.content.Context

import com.zhuinden.simplestackdemoexamplefragments.application.injection.DaggerSingletonComponent
import com.zhuinden.simplestackdemoexamplefragments.application.injection.SingletonComponent

/**
 * Created by Zhuinden on 2018. 08. 20.
 */

class CustomApplication : Application() {
    private var component: SingletonComponent? = null

    companion object {
        private var INSTANCE: CustomApplication? = null

        @JvmStatic
        fun getComponent() = get().component!!

        @JvmStatic
        fun get(context: Context): CustomApplication =
            context.applicationContext as CustomApplication

        @JvmStatic
        fun get(): CustomApplication = INSTANCE!!
    }

    override fun onCreate() {
        super.onCreate()
        INSTANCE = this
        component = DaggerSingletonComponent.create()
    }
}
